### PR TITLE
feat(moco): Add MOCO extension

### PR DESCRIPTION
**Pull request by eviscares**
## Description

Allows to interface with the timetracking part of [https://www.mocoapp.com/](https://www.mocoapp.com/).
Features:
- Checking of todays activities
- Checking of recent activities
- Logging of new activities
- Editing of logged activities
- Stopping & Starting the timer on an activity if it is from today.

## Screencast
https://drive.google.com/file/d/1gYNs-TK1CbbCPIPN7SIdQ2m1ULXGVjwz/view?usp=sharing

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by eviscares** [comment]
The pipeline fail seems weird to me.
It runs locally, and `npm run build` also works without any issue.

File: extensions/moco/src/commands/tasks/components/TaskListItem.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import { ActivityStart } from "../../activities/components/ActivityStart";
+
+export const TaskListItem = ({ task }: { task: Task }) => {
+  return (
+    <List.Item
+      key={task.id}
+      title={task.name}
+      actions={
+        <ActionPanel>
+          <Action.Push title={"Start an activity for this task"} target={<ActivityStart task={task} />} />
```
Comment (pernielsentikaer) [inline extensions/moco/src/commands/tasks/components/TaskListItem.tsx:12]:
```suggestion
          <Action.Push title={"Start Activity"} target={<ActivityStart task={task} />} />
```

File: extensions/moco/src/commands/activities/components/ActivityEdit.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    }
+  }
+
+  return (
+    <Form
+      navigationTitle={`Editing on ${activity.project.name}/${activity.task.name}`}
+      actions={
+        <ActionPanel>
+          <Action.SubmitForm
+            title="Save changes"
```
Comment (pernielsentikaer) [inline extensions/moco/src/commands/activities/components/ActivityEdit.tsx:36]:
```suggestion
            title="Save Changes"
```

File: extensions/moco/src/commands/activities/components/ActivityEdit.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        onChange={dropDescriptionErrorIfNeeded}
+        onBlur={(event) => {
+          if (event.target.value?.length == 0) {
+            setDescriptionError("The field should't be empty!");
+          } else {
+            dropDescriptionErrorIfNeeded();
+          }
+        }}
+      />
+      <Form.DatePicker id="date" title="For which day is the booking?" defaultValue={new Date(activity.date)} />
```
Comment (pernielsentikaer) [inline extensions/moco/src/commands/activities/components/ActivityEdit.tsx:72]:
```suggestion
      <Form.DatePicker id="date" title="Booking Date" defaultValue={new Date(activity.date)} />
```

File: extensions/moco/src/commands/activities/components/ActivityEdit.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+      <Form.TextField
+        id="hours"
+        title="How many hours did you work?"
```
Comment (pernielsentikaer) [inline extensions/moco/src/commands/activities/components/ActivityEdit.tsx:75]:
```suggestion
        title="Hours Worked"
```

File: extensions/moco/src/commands/activities/components/ActivityStart.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+            setDescriptionError("The field should't be empty!");
+          } else {
+            dropDescriptionErrorIfNeeded();
+          }
+        }}
+      />
+      <Form.DatePicker
+        id="date"
+        type={Form.DatePicker.Type.Date}
+        title="For which day is the booking?"
```
Comment (pernielsentikaer) [inline extensions/moco/src/commands/activities/components/ActivityStart.tsx:51]:
```suggestion
        title="Booking Date"
```

File: extensions/moco/src/commands/activities/components/ActivityStart.tsx [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+      <Form.TextField
+        id="hours"
+        onChange={(event) => {
+          if (event.length == 0) {
+            setStartTimer(true);
+          } else {
+            setStartTimer(false);
+          }
+        }}
+        title="How many hours did you work?"
```
Comment (pernielsentikaer) [inline extensions/moco/src/commands/activities/components/ActivityStart.tsx:63]:
```suggestion
        title="Hours Worked"
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: extensions/moco/src/commands/activities/components/ActivityEdit.tsx (hunk 2 above)
Comment (eviscares) [reply extensions/moco/src/commands/activities/components/ActivityEdit.tsx]: Changed

File: extensions/moco/src/commands/activities/components/ActivityEdit.tsx (hunk 3 above)
Comment (eviscares) [reply extensions/moco/src/commands/activities/components/ActivityEdit.tsx]: Changed

File: extensions/moco/src/commands/activities/components/ActivityEdit.tsx (hunk 4 above)
Comment (eviscares) [reply extensions/moco/src/commands/activities/components/ActivityEdit.tsx]: Changed

File: extensions/moco/src/commands/activities/components/ActivityStart.tsx (hunk 5 above)
Comment (eviscares) [reply extensions/moco/src/commands/activities/components/ActivityStart.tsx]: Changed

File: extensions/moco/src/commands/activities/components/ActivityStart.tsx (hunk 6 above)
Comment (eviscares) [reply extensions/moco/src/commands/activities/components/ActivityStart.tsx]: Changed

**Comment by eviscares** [comment]
Thanks for the Review and Feedback. I rephrased according to your suggestion.

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋

Looks good to me, approved 🔥

**Pull request merged**